sources/zafira-services/src/main/java/com/qaprosoft/zafira/service/integration/core/IntegrationTenancyStorage.java

```diff
@@ -104,10 +105,18 @@ public void initDb() {
 
     @RabbitListener(queues = "#{settingsQueue.name}")
     public void process(Message message) {
-        ReinitEventMessage rm = new Gson().fromJson(new String(message.getBody()), ReinitEventMessage.class);
-        if (!eventPushService.isSettingQueueConsumer(message)) {
-            Integration integration = integrationService.retrieveById(rm.getIntegrationId());
-            integrationInitializer.initIntegration(integration, rm.getTenancy());
+        try {
+            ReinitEventMessage rm = new Gson().fromJson(new String(message.getBody()), ReinitEventMessage.class);
+            if (!eventPushService.isSettingQueueConsumer(message)) {
+                TenancyContext.setTenantName(rm.getTenancy());
+
+                Integration integration = integrationService.retrieveById(rm.getIntegrationId());
+                integrationInitializer.initIntegration(integration, rm.getTenancy());
+
+                TenancyContext.setTenantName(null);
+            }
+        } catch (Exception e) {
+            LOGGER.error("Unable to initialize adapter. " + e.getMessage(), e);
```

Comment:
What exactly might cause an exception here? If message is not malformed (can we add a separate catch?) can we obtain from message and log something useful for us like integration id at least?